src/main/java/org/opentripplanner/routing/api/request/preference/VehicleParkingPreferences.java

```diff
@@ -0,0 +1,228 @@
+package org.opentripplanner.routing.api.request.preference;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import org.opentripplanner.framework.model.Cost;
+import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.routing.api.request.request.filter.VehicleParkingFilter;
+import org.opentripplanner.routing.api.request.request.filter.VehicleParkingFilterRequest;
+
+/**
+ * The parking preferences contain preferences for car and bicycle parking. These preferences
+ * include filtering, preference and realtime usage.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE.
+ */
+public final class VehicleParkingPreferences implements Serializable {
+
+  public static final VehicleParkingPreferences DEFAULT = new VehicleParkingPreferences();
+  private final Cost unpreferredVehicleParkingTagCost;
+  private final boolean useAvailabilityInformation;
+  private final VehicleParkingFilterRequest filter;
+  private final VehicleParkingFilterRequest preferred;
+  private final Duration parkTime;
+  private final Cost parkCost;
+
+  /** Create a new instance with default values. */
+  private VehicleParkingPreferences() {
+    this.unpreferredVehicleParkingTagCost = Cost.costOfMinutes(5);
+    this.useAvailabilityInformation = true;
+    this.filter = VehicleParkingFilterRequest.empty();
+    this.preferred = VehicleParkingFilterRequest.empty();
+    this.parkTime = Duration.ofMinutes(1);
+    this.parkCost = Cost.costOfMinutes(2);
+  }
+
+  private VehicleParkingPreferences(Builder builder) {
+    this.unpreferredVehicleParkingTagCost = builder.unpreferredVehicleParkingTagCost;
+    this.useAvailabilityInformation = builder.useAvailabilityInformation;
+    this.filter =
+      new VehicleParkingFilterRequest(
+        builder.bannedVehicleParkingTags,
+        builder.requiredVehicleParkingTags
+      );
+    this.preferred =
+      new VehicleParkingFilterRequest(
+        builder.notPreferredVehicleParkingTags,
+        builder.preferredVehicleParkingTags
+      );
+    this.parkTime = builder.parkTime;
+    this.parkCost = builder.parkCost;
+  }
+
+  public static VehicleParkingPreferences.Builder of() {
+    return new Builder(DEFAULT);
+  }
+
+  /**
+   * What cost is applied to using parking that is not preferred.
+   */
+  public Cost unpreferredVehicleParkingTagCost() {
+    return unpreferredVehicleParkingTagCost;
+  }
+
+  /**
+   * Should availability information be used during routing.
+   */
+  public boolean useAvailabilityInformation() {
+    return useAvailabilityInformation;
+  }
+
+  /**
+   * Parking containing select filters must only be usable and parking containing with not filters
+   * cannot be used.
+   */
+  public VehicleParkingFilterRequest filter() {
```

Comment:
This is no longer part of the request, so it need a new name. I suggest just dropping the `Request`, and we will deal with the naming when splitting the data from the filtering implementation.

Reply:
Issue is that this class contains VehicleParkingFilter objects already so can't use that name unless we first rename it. I couldn't really think of a good name for this. We can come up with a solution in the dev meeting.